### ISSUE TITLE
4.0b1 -  Peripherals & Devices

### ISSUE THREAD

**Issue by bluecursor** [dirkwhoffmann/virtualc64#610]
Unable to save Peripherals defaults.
Also Devices is recognizing my trackball mouse as a joystick

![Screen Shot 2021-01-13 at 5 01 29 PM](https://user-images.githubusercontent.com/31462938/104515763-39b2cb80-55c1-11eb-87f2-d7acbe4e2b96.png)

**Comment by dirkwhoffmann** [comment]
Thanks for reporting this. The recognition bug is here:

```
       // Prepare to accept HID devices
        let deviceCriteria = [
            [
                kIOHIDDeviceUsagePageKey: kHIDPage_GenericDesktop,
                kIOHIDDeviceUsageKey: kHIDUsage_GD_Joystick
            ],
            [
                kIOHIDDeviceUsagePageKey: kHIDPage_GenericDesktop,
                kIOHIDDeviceUsageKey: kHIDUsage_GD_GamePad
            ],
            [
                kIOHIDDeviceUsagePageKey: kHIDPage_GenericDesktop,
                kIOHIDDeviceUsageKey: kHIDUsage_GD_MultiAxisController
            ],
            [
                kIOHIDDeviceUsagePageKey: kHIDPage_GenericDesktop,
                kIOHIDDeviceUsageKey: kHIDUsage_GD_Mouse
            ]
        ]
```

In contrast to vAmiga, VirtualC64 is not supposed to accept devices with usage type `kHIDUsage_GD_Mouse`. Hence, the last key pair needs to be deleted. 

Your mouse should not show up as a separate HID device at all, because mice are handled by macOS itself. vAmiga is different in this aspect, because it supports two mice. VirtualC64 only supports one mouse which makes the implementation much easier. 

The "mouse type" is recognized correctly though. It's the number printed next to "Usage":

<img width="675" alt="Bildschirmfoto 2021-01-14 um 08 02 03" src="https://user-images.githubusercontent.com/12561945/104556684-54785500-5640-11eb-8bc2-410e072cc72f.png">

In the next version, I'll print a descriptive string instead of the raw number:

<img width="588" alt="Bildschirmfoto 2021-01-14 um 08 09 51" src="https://user-images.githubusercontent.com/12561945/104556751-69ed7f00-5640-11eb-9915-f5000e14cc64.png">

Could you doublecheck whether saving really doesn't work? The emulator is supposed to remember the setting on a per device basis. Hence, the previous setting is only restored when the same device is plugged in again. Internally, the emulator saves a string map into the user default storage which looks like this on my machine:

`DeviceDatabase.61::init(): ["1035": ["25907": ["H": "1", "L": "1"]]]`

It's a nested map of the form `( VendorID -> (ProductID -> Dictionary) )`.

Important: The map is written to the user defaults storage when the Preferences window is exited by clicking OK.  If you close the window by clicking the red icon, it won't be saved. I am unsure if this is the correct behavior. Somebody needs to peek into the Apple UX guidelines 🙄.

**Comment by bluecursor** [comment]
In my test, I am trying to save enable Drive 9 and Port 2 Keypad Joystick and will not save when restarting fresh.

**Comment by dirkwhoffmann** [comment]
Should be fixed in v4.0b2.
